system/services/presideObjects/PresideObjectService.cfc

```diff
@@ -2189,6 +2190,53 @@ component displayName="Preside Object Service" {
 		return fields;
 	}
 
+	public array function parseSelectFieldsForHavingClause(
+		  required array  selectFields
+		, required array  extraFilters
+		, required string objectName
+	) {
+		var extraFilters = arguments.extraFilters;
+		var selectFields = Duplicate( arguments.selectFields );
+		var props        = getObjectProperties( arguments.objectName );
+
+		for ( var filter in extraFilters ) {
+			if( Len( Trim( filter.having ?: "" ) ) ) {
+				var fields = filter.havingfields ?: ListToArray( filter.propertyName ?: "" );
+
+				for ( var field in fields ) {
+					var propertyLen = ListLen( field, "." );
+
+					if ( propertyLen == 1 ) {
+						if( !ArrayFindNoCase( selectFields, field ) ) {
+							field = "#arguments.objectName#.#field#";
+						}
+					}
+					else if (propertyLen == 2 ) {
+						var prefix       = ListFirst( field, "." );
+						var propertyName = ListLast( field, "." );
+						var prop         = props[ propertyName ] ?: {};
```

Comment:
I believe there may be a strong possibilty that the property does not belong to the object directly. i.e. it could be `related$object$something.propertyName`.

We have similar logic for getting the property from source object and field path in this service and we should re-use it here to ensure we are reading the property from the correct place.